simpleclient_common/src/main/java/io/prometheus/client/exporter/common/TextFormat.java

```diff
@@ -2,37 +2,108 @@
 
 import java.io.IOException;
 import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 
 import io.prometheus.client.Collector;
 
 public class TextFormat {
   /**
-   * Content-type for text version 0.0.4.
+   * Content-type for Prometheus text version 0.0.4.
    */
   public final static String CONTENT_TYPE_004 = "text/plain; version=0.0.4; charset=utf-8";
 
+  /**
+   * Content-type for Openmetrics text version 1.0.0.
+   */
+  public final static String CONTENT_TYPE_OPENMETRICS_100 = "application/openmetrics-text; version=1.0.0; charset=utf-8";
+
+  /**
+   * Return the content type that should be used for a given Accept HTTP header.
+   */
+  public static String chooseContentType(String acceptHeader) {
+    if (acceptHeader == null) {
+      return CONTENT_TYPE_004;
+    }
+
+    for (String accepts : acceptHeader.split(",")) {
+      if ("application/openmetrics-text".equals(accepts.split(";")[0].trim())) {
+        return CONTENT_TYPE_OPENMETRICS_100;
+      }
+    }
+
+    return CONTENT_TYPE_004;
+  }
+
+  /**
+   * Write out the given MetricFamilySamples in a format per the contentType.
+   */
+  public static void writeFormat(String contentType, Writer writer, Enumeration<Collector.MetricFamilySamples> mfs) throws IOException {
+    if (CONTENT_TYPE_004.equals(contentType)) {
+        write004(writer, mfs);
+        return;
+    }
+    if (CONTENT_TYPE_OPENMETRICS_100.equals(contentType)) {
+        writeOpenMetrics100(writer, mfs);
+        return;
+    }
+    throw new IllegalArgumentException("Unknown contentType " + contentType);
+  }
+
   /**
    * Write out the text version 0.0.4 of the given MetricFamilySamples.
    */
   public static void write004(Writer writer, Enumeration<Collector.MetricFamilySamples> mfs) throws IOException {
+    Map<String, Collector.MetricFamilySamples> omFamilies = new TreeMap<String, Collector.MetricFamilySamples>();
     /* See http://prometheus.io/docs/instrumenting/exposition_formats/
      * for the output format specification. */
     while(mfs.hasMoreElements()) {
       Collector.MetricFamilySamples metricFamilySamples = mfs.nextElement();
+      String name = metricFamilySamples.name;
       writer.write("# HELP ");
-      writer.write(metricFamilySamples.name);
+      writer.write(name);
+      if (metricFamilySamples.type == Collector.Type.COUNTER) {
+        writer.write("_total");
+      }
```

Comment:
Hm is Prometheus text version output also meant to have the suffixes appended? I thought only OpenMetrics format would get that?

Reply:
Yes, no matter which format is negotiated the exact same samples should end up inside Prometheus.